File: 2024/03B.py
import sys
import re

def solve(input):
    result = 0
    
    parts = input.split('do()')
    for part in parts:
        part = part.split("don't()")[0]
        result += sum([int(a) * int(b) for (a,b) in re.findall(r'mul\((\d+),(\d+)\)', part)])

    return result
        
if __name__ == '__main__':
        print(solve(open(sys.argv[1]).read()))

File: 2024/03B.js
function solve(input) {
  let result = 0;
  for (let part of input.split('do()')) {
    let str = part.split("don't()")[0];
    for (let match of str.matchAll(/mul\((\d+),(\d+)\)/g)) {
      result += match[1] * match[2];
    }
  }
  return result;
}

console.log(solve(require('node:fs').readFileSync(process.argv[2], 'utf8').trim()));